internal/comparetest/logs.go

```diff
@@ -132,34 +124,32 @@ func CompareResourceLogs(expected, actual plog.ResourceLogs) error {
 // CompareLogRecordSlices compares each part of two given LogRecordSlices and returns
 // an error if they don't match. The error describes what didn't match.
 func CompareLogRecordSlices(expected, actual plog.LogRecordSlice) error {
-	exp, act := plog.NewLogRecordSlice(), plog.NewLogRecordSlice()
-	expected.CopyTo(exp)
-	actual.CopyTo(act)
-
-	if exp.Len() != act.Len() {
-		return fmt.Errorf("number of log records does not match expected: %d, actual: %d", exp.Len(), act.Len())
+	if expected.Len() != actual.Len() {
+		return fmt.Errorf("number of log records does not match expected: %d, actual: %d", expected.Len(), actual.Len())
 	}
 
-	exp.Sort(sortLogRecordSlice)
-	act.Sort(sortLogRecordSlice)
-
-	numLogRecords := exp.Len()
+	numLogRecords := expected.Len()
 
 	// Keep track of matching records so that each record can only be matched once
 	matchingLogRecords := make(map[plog.LogRecord]plog.LogRecord, numLogRecords)
 
 	var errs error
+	var outOfOrderErr error
 	for e := 0; e < numLogRecords; e++ {
-		elr := exp.At(e)
+		elr := expected.At(e)
 		var foundMatch bool
 		for a := 0; a < numLogRecords; a++ {
-			alr := act.At(a)
+			alr := actual.At(a)
 			if _, ok := matchingLogRecords[alr]; ok {
 				continue
 			}
 			if reflect.DeepEqual(elr.Attributes().AsRaw(), alr.Attributes().AsRaw()) {
 				foundMatch = true
 				matchingLogRecords[alr] = elr
+				if e != a && outOfOrderErr == nil {
+					outOfOrderErr = fmt.Errorf("LogRecord with attributes %v expected at index %d, "+
```

Comment:
Is this type of error deduplicated because it may be too noisy?

Reply:
Yes, but we can show them all if you think it's a better approach

Reply:
I'm fine with it this way, just wanted to make sure I understood the motivation

Reply:
Maybe for consistency with extra/missing errors it's better to show all